### PR TITLE
[Merged by Bors] - `Gamepad` type is `Copy`; do not require / return references to it in `Gamepads` API

### PR DESCRIPTION
**Pull request by alice-i-cecile**
# Objective

- The `Gamepad` type is a tiny value-containing type that implements `Copy`.
- By convention, references to `Copy` types should be avoided, as they can introduce overhead and muddle the semantics of what's going on.
- This allows us to reduce boilerplate reference manipulation and lifetimes in user facing code.

## Solution

- Make assorted methods on `Gamepads` take / return a raw `Gamepad`, rather than `&Gamepad`.

## Migration Guide

- `Gamepads::iter` now returns an iterator of `Gamepad`. rather than an iterator of `&Gamepad`.
- `Gamepads::contains` now accepts a `Gamepad`, rather than a `&Gamepad`.

### CODE REVIEW

**Comment by Nilirad** [comment]
Labeled as breaking change since it changes function signatures.

**Review by hymm — APPROVED** [review]
LGTM

File: crates/bevy_input/src/gamepad.rs [hunk 1]
```diff
@@ -46,24 +46,24 @@ pub struct Gamepads {
...
-        self.gamepads.contains(gamepad)
+    /// Returns true if [`Gamepads`] contains a [`Gamepad`].
+    pub fn contains(&self, gamepad: Gamepad) -> bool {
+        self.gamepads.contains(&gamepad)
     }
 
-    /// An iterator visiting all connected [`Gamepad`]s in arbitrary order.
-    pub fn iter(&self) -> impl Iterator<Item = &Gamepad> + '_ {
-        self.gamepads.iter()
+    /// Iterates over registered [`Gamepad`]s
```
Comment (alice-i-cecile) [inline crates/bevy_input/src/gamepad.rs:54]:
```suggestion
    /// Returns an iterator over registered [`Gamepad`]s in an arbitrary order
```

File: crates/bevy_input/src/gamepad.rs [hunk 2]
```diff
@@ -46,24 +46,24 @@ pub struct Gamepads {
 }
 
 impl Gamepads {
-    /// Returns `true` if the `gamepad` is connected.
-    pub fn contains(&self, gamepad: &Gamepad) -> bool {
-        self.gamepads.contains(gamepad)
+    /// Returns true if [`Gamepads`] contains a [`Gamepad`].
```
Comment (alice-i-cecile) [inline crates/bevy_input/src/gamepad.rs:49]:
```suggestion
    /// Returns true if the `gamepad` is connected.
```

File: crates/bevy_input/src/gamepad.rs [hunk 3]
```diff
@@ -46,24 +46,24 @@ pub struct Gamepads {
...
 
     /// Registers the `gamepad` marking it as connected.
     fn register(&mut self, gamepad: Gamepad) {
         self.gamepads.insert(gamepad);
     }
 
-    /// Deregisters the `gamepad` marking it as disconnected.
-    fn deregister(&mut self, gamepad: &Gamepad) {
-        self.gamepads.remove(gamepad);
+    /// Deregisters the [`Gamepad`], marking it as disconnected.
```
Comment (alice-i-cecile) [inline crates/bevy_input/src/gamepad.rs:64]:
```suggestion
    /// Deregisters the `gamepad`, marking it as disconnected.
```

File: crates/bevy_input/src/gamepad.rs [hunk 4]
```diff
@@ -46,24 +46,24 @@ pub struct Gamepads {
 }
 
 impl Gamepads {
-    /// Returns `true` if the `gamepad` is connected.
-    pub fn contains(&self, gamepad: &Gamepad) -> bool {
-        self.gamepads.contains(gamepad)
+    /// Returns true if the `gamepad` is connected.
```
Comment (alice-i-cecile) [inline crates/bevy_input/src/gamepad.rs:49]:
```suggestion
    /// Returns `true` if the `gamepad` is connected.
```

File: crates/bevy_input/src/gamepad.rs [hunk 5]
```diff
@@ -47,23 +47,23 @@ pub struct Gamepads {
...
-    pub fn contains(&self, gamepad: &Gamepad) -> bool {
-        self.gamepads.contains(gamepad)
+    pub fn contains(&self, gamepad: Gamepad) -> bool {
+        self.gamepads.contains(&gamepad)
     }
 
-    /// An iterator visiting all connected [`Gamepad`]s in arbitrary order.
-    pub fn iter(&self) -> impl Iterator<Item = &Gamepad> + '_ {
-        self.gamepads.iter()
+    /// Returns an iterator over registered [`Gamepad`]s in an arbitrary order
```
Comment (devil-ira) [inline crates/bevy_input/src/gamepad.rs:54]: Unyeet period.
```suggestion
    /// Returns an iterator over registered [`Gamepad`]s in an arbitrary order.
```

**Review by devil-ira — APPROVED** [review]

File: crates/bevy_input/src/gamepad.rs [hunk 6]
```diff
@@ -47,23 +47,23 @@ pub struct Gamepads {
...
 
-    /// An iterator visiting all connected [`Gamepad`]s in arbitrary order.
-    pub fn iter(&self) -> impl Iterator<Item = &Gamepad> + '_ {
-        self.gamepads.iter()
+    /// Returns an iterator over registered [`Gamepad`]s in an arbitrary order.
+    pub fn iter(&self) -> impl Iterator<Item = Gamepad> + '_ {
+        self.gamepads.iter().copied()
     }
 
     /// Registers the `gamepad` marking it as connected.
```
Comment (TimJentzsch) [inline crates/bevy_input/src/gamepad.rs:59]: Let's be consistent and add a comma here too :)
```suggestion
    /// Registers the `gamepad`, marking it as connected.
```

**Review by TimJentzsch — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
bors retry

**Pull request closed without merging**